// @strapi/design-system/dist/Textarea/Textarea.js
import r from "react";
import e from "prop-types";
import h from "styled-components";
import { TextareaInput as u } from "./TextareaInput.js";
import { Field as g } from "../Field/Field.js";
import { FieldLabel as v } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as x } from "../Field/FieldHint.js";
import { FieldError as E } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Flex as T } from "../Flex/Flex.js";
import { useId as b } from "../helpers/useId.js";
import { Stack as y } from "../Stack/Stack.js";
const F = h.div`
  & textarea {
    // TODO: remove when we'll have fonts in the theme
    height: ${80 / 16}rem;
    line-height: ${20 / 16}rem;
  }

  & textarea::placeholder {
    font-weight: 400;
    font-size: ${14 / 16}rem;
    line-height: 1.43;
    color: ${({ theme: t }) => t.colors.neutral500};
    opacity: 1;
  }
`, i = r.forwardRef(
  ({ name: t, hint: a, error: l, label: o, children: n, labelAction: m, id: d, required: p, ...c }, s) => {
    const f = b(d);
    return /* @__PURE__ */ r.createElement(F, null, /* @__PURE__ */ r.createElement(g, { name: t, hint: a, error: l, id: f, required: p }, /* @__PURE__ */ r.createElement(y, { spacing: 1 }, o && /* @__PURE__ */ r.createElement(T, null, /* @__PURE__ */ r.createElement(v, { action: m }, o)), /* @__PURE__ */ r.createElement(u, { ref: s, as: "textarea", value: n, ...c }), /* @__PURE__ */ r.createElement(x, null), /* @__PURE__ */ r.createElement(E, null))));
  }
);
i.displayName = "Textarea";
i.defaultProps = {
  "aria-label": void 0,
  label: void 0,
  labelAction: void 0,
  error: void 0,
  hint: void 0,
  id: void 0,
  children: "",
  required: !1
};
i.propTypes = {
  "aria-label": e.string,
  children: e.string,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.string,
  label: e.string,
  labelAction: e.element,
  name: e.string.isRequired,
  required: e.bool
};
export {
  i as Textarea
};
