// @strapi/design-system/dist/JSONInput/JSONInput.js
import { jsx as e, jsxs as I } from "react/jsx-runtime";
import { useRef as l, useEffect as J } from "react";
import { json as j, jsonParseLinter as H } from "@codemirror/lang-json";
import { useCodeMirror as L } from "@uiw/react-codemirror";
import { JSONInputContainer as z } from "./JSONInputContainer.js";
import { markField as A, addMarks as B, lineHighlightMark as R, filterMarks as q } from "./utils/decorationExtension.js";
import { Field as y } from "../Field/Field.js";
import { FieldLabel as P } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as V } from "../Field/FieldHint.js";
import { FieldError as D } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Stack as G } from "../Stack/Stack.js";
const nr = ({
  label: a,
  error: d,
  hint: p,
  labelAction: g,
  value: E = "",
  required: S = !1,
  disabled: u = !1,
  onChange: k = () => null,
  ...C
}) => {
  const o = l(), i = l(), c = l(), b = Boolean(d), F = (r) => {
    const t = i.current?.doc;
    if (t) {
      const {
        text: s,
        to: n
      } = t.line(r), m = n - s.trimStart().length;
      n > m && c.current?.dispatch({
        effects: B.of([R.range(m, n)])
      });
    }
  }, M = () => {
    const r = i.current?.doc;
    if (r) {
      const t = r.length || 0;
      c.current?.dispatch({
        effects: q.of((s, n) => n <= 0 || s >= t)
      });
    }
  }, f = ({
    state: r,
    view: t
  }) => {
    c.current = t, i.current = r, M();
    const n = H()(t);
    n.length && F(r.doc.lineAt(n[0].from).number);
  }, x = (r, t) => {
    f(t), k(r);
  }, N = (r, t) => {
    c.current = r, i.current = t, f({
      view: r,
      state: t
    });
  }, {
    setContainer: h
  } = L({
    value: E,
    onCreateEditor: N,
    container: o.current,
    editable: !u,
    extensions: [j(), A],
    onChange: x,
    theme: "dark",
    basicSetup: {
      lineNumbers: !0,
      bracketMatching: !0,
      closeBrackets: !0,
      indentOnInput: !0,
      syntaxHighlighting: !0,
      highlightSelectionMatches: !0,
      tabSize: 2
    }
  }), O = () => {
    u || o.current?.children[0].children[1].children[1]?.focus();
  };
  return J(() => {
    const r = o.current;
    r && h(r);
  }, [h]), e(y, {
    error: d,
    hint: p,
    required: S,
    children: I(G, {
      spacing: 1,
      children: [a && e(P, {
        onClick: O,
        action: g,
        children: a
      }), e(z, {
        ref: o,
        hasError: b,
        alignItems: "stretch",
        fontSize: 2,
        hasRadius: !0,
        ...C
      }), e(D, {}), e(V, {})]
    })
  });
};
export {
  nr as JSONInput
};
