// @strapi/design-system/dist/Checkbox/Checkbox.js
import r from "react";
import e from "prop-types";
import s from "styled-components";
import { BaseCheckbox as c } from "../BaseCheckbox/BaseCheckbox.js";
import { Box as f } from "../Box/Box.js";
import { Field as u } from "../Field/Field.js";
import "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as b } from "../Field/FieldHint.js";
import { FieldError as y } from "../Field/FieldError.js";
import { useField as x } from "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { useId as E } from "../helpers/useId.js";
import { Stack as g } from "../Stack/Stack.js";
import { Typography as h } from "../Typography/Typography.js";
const k = s(h)`
  display: flex;
  align-items: flex-start;
  * {
    cursor: ${({ disabled: o }) => o ? "not-allowed" : "pointer"};
  }
`, T = (o) => {
  const { id: t } = x();
  return /* @__PURE__ */ r.createElement(c, { id: t, ...o });
}, a = ({ children: o, disabled: t, id: p, hint: m, error: l, ...d }) => {
  const i = E(p);
  let n;
  return l ? n = `${i}-error` : m && (n = `${i}-hint`), /* @__PURE__ */ r.createElement(u, { id: i, hint: m, error: l }, /* @__PURE__ */ r.createElement(g, { spacing: 1 }, /* @__PURE__ */ r.createElement(k, { as: "label", textColor: "neutral800", disabled: t }, /* @__PURE__ */ r.createElement(T, { disabled: t, "aria-describedby": n, ...d }), /* @__PURE__ */ r.createElement(f, { paddingLeft: 2 }, o)), /* @__PURE__ */ r.createElement(b, null), /* @__PURE__ */ r.createElement(y, null)));
};
a.defaultProps = {
  disabled: !1,
  id: void 0,
  error: void 0,
  hint: void 0
};
a.propTypes = {
  children: e.node.isRequired,
  disabled: e.bool,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.oneOfType([e.string, e.number])
};
export {
  a as Checkbox
};
