// @strapi/design-system/dist/NumberInput/NumberInput.js
import t, { useRef as g } from "react";
import { NumberParser as L, NumberFormatter as T } from "@internationalized/number";
import { CarretDown as v } from "@strapi/icons";
import r from "prop-types";
import z from "styled-components";
import { Field as B } from "../Field/Field.js";
import { FieldLabel as U } from "../Field/FieldLabel.js";
import { FieldInput as _ } from "../Field/FieldInput.js";
import { FieldHint as j } from "../Field/FieldHint.js";
import { FieldError as H } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { getDefaultLocale as M } from "../helpers/getDefaultLocale.js";
import { KeyboardKeys as N } from "../helpers/keyboardKeys.js";
import { useId as W } from "../helpers/useId.js";
import { useControllableState as X } from "../hooks/useControllableState.js";
import { Icon as y } from "../Icon/Icon.js";
import { Stack as Y } from "../Stack/Stack.js";
import { sizes as G } from "../themes/sizes.js";
const h = z.button`
  display: flex;
  height: 1rem;
  align-items: ${({ reverse: n }) => n ? "flex-end" : "flex-start"};
  transform: translateY(${({ reverse: n }) => n ? "-2px" : "2px"});
  cursor: ${({ disabled: n }) => n ? "not-allowed" : void 0};
  svg {
    display: block;
    height: ${4 / 16}rem;
    transform: ${({ reverse: n }) => n ? "rotateX(180deg)" : void 0};
  }
`, J = "", u = t.forwardRef(
  ({
    size: n,
    startAction: w,
    name: E,
    hint: I,
    error: A,
    label: d,
    labelAction: V,
    locale: x,
    id: D,
    onValueChange: F,
    value: k,
    step: i,
    required: C,
    disabled: m,
    ...P
  }, R) => {
    const O = W(D), p = x || M(), s = g(new L(p, { style: "decimal" })), c = g(new T(p, { maximumFractionDigits: 20 })), [a, S] = X({
      prop(e) {
        const o = String(k);
        return isNaN(o) || o !== e && e !== "" ? e : o;
      },
      defaultProp: J,
      onChange(e) {
        const o = s.current.parse(e);
        F(isNaN(o) ? void 0 : o);
      }
    }), l = (e) => {
      S(String(e));
    }, $ = ({ target: { value: e } }) => {
      s.current.isValidPartialNumber(e) && l(e);
    }, f = () => {
      if (!a) {
        l(i);
        return;
      }
      const e = s.current.parse(a), o = isNaN(e) ? i : e + i;
      l(c.current.format(o));
    }, b = () => {
      if (!a) {
        l(-i);
        return;
      }
      const e = s.current.parse(a), o = isNaN(e) ? -i : e - i;
      l(c.current.format(o));
    }, q = (e) => {
      if (!m)
        switch (e.key) {
          case N.DOWN: {
            e.preventDefault(), b();
            break;
          }
          case N.UP: {
            e.preventDefault(), f();
            break;
          }
        }
    }, K = () => {
      if (a) {
        const e = s.current.parse(a), o = isNaN(e) ? "" : c.current.format(e);
        l(o);
      }
    };
    return /* @__PURE__ */ t.createElement(B, { name: E, hint: I, error: A, id: O, required: C }, /* @__PURE__ */ t.createElement(Y, { spacing: 1 }, d && /* @__PURE__ */ t.createElement(U, { action: V }, d), /* @__PURE__ */ t.createElement(
      _,
      {
        ref: R,
        startAction: w,
        disabled: m,
        type: "text",
        inputmode: "decimal",
        onChange: $,
        onKeyDown: q,
        onBlur: K,
        value: a,
        size: n,
        endAction: /* @__PURE__ */ t.createElement(t.Fragment, null, /* @__PURE__ */ t.createElement(
          h,
          {
            disabled: m,
            "aria-hidden": !0,
            reverse: !0,
            onClick: f,
            tabIndex: -1,
            type: "button",
            "data-testid": "ArrowUp"
          },
          /* @__PURE__ */ t.createElement(y, { as: v, color: "neutral500" })
        ), /* @__PURE__ */ t.createElement(
          h,
          {
            disabled: m,
            "aria-hidden": !0,
            onClick: b,
            tabIndex: -1,
            type: "button",
            "data-testid": "ArrowDown"
          },
          /* @__PURE__ */ t.createElement(y, { as: v, color: "neutral500" })
        )),
        ...P
      }
    ), /* @__PURE__ */ t.createElement(j, null), /* @__PURE__ */ t.createElement(H, null)));
  }
);
u.displayName = "NumberInput";
u.defaultProps = {
  "aria-label": void 0,
  disabled: !1,
  error: void 0,
  hint: void 0,
  id: void 0,
  label: void 0,
  labelAction: void 0,
  locale: void 0,
  required: !1,
  size: "M",
  startAction: void 0,
  step: 1,
  value: void 0
};
u.propTypes = {
  "aria-label": r.string,
  disabled: r.bool,
  error: r.string,
  hint: r.oneOfType([r.string, r.node, r.arrayOf(r.node)]),
  id: r.string,
  label: r.string,
  labelAction: r.element,
  locale: r.string,
  name: r.string.isRequired,
  onValueChange: r.func.isRequired,
  required: r.bool,
  size: r.oneOf(Object.keys(G.input)),
  startAction: r.element,
  step: r.number,
  value: r.number
};
export {
  u as NumberInput
};
